api/types/role.go

```diff
@@ -71,6 +71,9 @@ type Role interface {
 	// SetMetadata sets role metadata
 	SetMetadata(meta Metadata)
 
+	// SetSpec sets role spec
+	SetSpec(spec RoleSpecV6)
```

Comment:
I assume you added this for role updates. I don't think it's necessary and I don't know how flexible that is given we bake-in the spec version. You also bypass spec validation. You can create a new role and the set existing role metadata. Pseudocode:

```go
existingRole := getExistingRole()
newRole, _ := types.NewRoleWithVersion(existingRole.GetName(), existingRole.GetVersion(), newSpec)
newRole.SetMetadata(existingRole.GetMetadata)
service.UpdateRole(ctx, newRole)
```